src/main/kotlin/no/skatteetaten/aurora/boober/service/DeployService.kt

```diff
@@ -180,62 +179,19 @@ class DeployService(
         val namespace = deploymentSpec.environment.namespace
         val name = deploymentSpec.name
 
-        val openShiftApplicationObjects: List<JsonNode> = openShiftObjectGenerator.generateApplicationObjects(deployId, deploymentSpec, provisioningResult)
-        val openShiftApplicationResponses: List<OpenShiftResponse> =
-                openShiftApplicationObjects.flatMap {
-                    val openShiftCommand = openShiftClient.createOpenShiftCommand(namespace, it, mergeWithExistingResource)
-                    if (updateRouteCommandWithChangedHostOrPath(openShiftCommand, deploymentSpec)) {
-                        val deleteCommand = openShiftCommand.copy(operationType = OperationType.DELETE)
-                        val createCommand = openShiftCommand.copy(operationType = OperationType.CREATE, payload = openShiftCommand.generated!!)
-                        listOf(deleteCommand, createCommand)
-                    } else {
-                        listOf(openShiftCommand)
-                    }
-                }.map { openShiftClient.performOpenShiftCommand(namespace, it) }
+        val openShiftApplicationObjects: List<OpenshiftCommand> = openShiftCommandBuilder.generateApplicationObjects(deployId, deploymentSpec, provisioningResult, mergeWithExistingResource)
```

Comment:
Do not see the need for intermediaate variable or type signature. it is obvious. Also line is probably a little bit to long :P

Reply:
Agreed. And fixed.